beacon-chain/operations/attestations/kv/forkchoice.go

```diff
@@ -8,7 +8,7 @@ import (
 
 // SaveForkchoiceAttestation saves an forkchoice attestation in cache.
 func (c *AttCaches) SaveForkchoiceAttestation(att ethpb.Att) error {
-	if att == nil {
+	if att == nil || att.IsNil() {
```

Comment:
If the `att == nil` wasn't working anyway should we not just replace it with `att.IsNil()`?
Same question for all other instances

```suggestion
	if att.IsNil() {
```

Reply:
I mention it in the description. you need both

> interfaces in Go: An interface is nil only when its type and value are both nil.
Typed nil in Interfaces: An interface holding a typed nil is not nil.
IsNil() is needed because we cannot rely on a simple nil check on the interface and must check if the underlying value is nil.

Reply:
You're right, I forgot when going through the code